ykman-gui/qml/PivSetupForMacOsView.qml

```diff
@@ -0,0 +1,138 @@
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Layouts 1.2
+import QtQuick.Controls.Material 2.2
+import "utils.js" as Utils
+
+ColumnLayout {
+
+    readonly property string authenticationName: 'Yubico PIV Authentication'
+    readonly property string encryptionName: 'Yubico PIV Encryption'
+    readonly property string algoritm: 'ECCP256'
+    property string expirationDate: Utils.formatDate(
+                                        getExpirationDateIn30years())
+    property bool isBusy
+
+    function getExpirationDateIn30years() {
+        var date = new Date()
+        date.setFullYear(date.getFullYear() + 30)
+        return date
+    }
+
+    function setupForMacOs(confirmedOverwrite) {
+        function _prompt_for_pin_and_key(pin, key) {
+            if (key) {
+                pivPinPopup.getPinAndThen(function (pin) {
+                    _finish(pin, key)
+                })
+            } else {
+                views.pivGetPinOrManagementKey(function (pin) {
+                    _finish(pin, false)
+                }, function (key) {
+                    _prompt_for_pin_and_key(false, key)
+                })
+            }
+        }
+
+        function _finish(pin, managementKey) {
+            isBusy = true
+            yubiKey.pivGenerateCertificate({
+                                               slotName: 'AUTHENTICATION',
+                                               algorithm: algoritm,
+                                               commonName: authenticationName,
+                                               expirationDate: expirationDate,
+                                               selfSign: true,
+                                               pin: pin,
+                                               keyHex: managementKey,
+                                               callback: function (resp) {
+                                                   pivError.showResponseError(
+                                                               resp)
+                                                   if (resp.success) {
+                                                       yubiKey.pivGenerateCertificate({
```

Comment:
I think this would sit better in its own separate `_finish` function. The two versions can probably also be refactored into simple calls to a generic one - it's only the slot and callback that differs, right?

Reply:
Done!